pkg/parser/parser.go

```diff
@@ -494,15 +508,27 @@ func (p *Parser) appendErrorForToken(message string, token *lexer.Token) {
 }
 
 func (p *Parser) parseBlock(scope *scope) *BlockStatement {
+	endTokens := map[lexer.TokenType]bool{lexer.END: true, lexer.EOF: true}
+	return p.parseBlockWithEndTokens(scope, endTokens)
+}
+
+func (p *Parser) parseIfBlock(scope *scope) *BlockStatement {
+	endTokens := map[lexer.TokenType]bool{lexer.END: true, lexer.EOF: true, lexer.ELSE: true}
+	return p.parseBlockWithEndTokens(scope, endTokens)
+}
+
+func (p *Parser) parseBlockWithEndTokens(scope *scope, endTokens map[lexer.TokenType]bool) *BlockStatement {
 	tok := p.cur
 	var stmts []Node
-	for p.cur.TokenType() != lexer.END && p.cur.TokenType() != lexer.EOF {
+	for !endTokens[p.cur.TokenType()] {
 		stmt := p.parseStatement(scope)
 		if stmt != nil {
 			stmts = append(stmts, stmt)
 		}
 	}
-	p.advancePastNL()
+	if len(stmts) == 0 {
+		p.appendErrorForToken("block statements must have at least one statement, none found", tok)
```

Comment:
I think using the term "block statements" is leaking the grammar into the error messages. This could just say "At least one statement is required jere" or something.

Reply:
yup, good idea. done.